x/common/testutil/sample.go

```diff
@@ -13,16 +15,52 @@ import (
 	tmdb "github.com/tendermint/tm-db"
 )
 
-// AccAddress Returns a sample account address (sdk.AccAddress)
+// AccAddress returns a sample address (sdk.AccAddress) created using secp256k1.
 // Note that AccAddress().String() can be used to get a string representation.
 func AccAddress() sdk.AccAddress {
+	_, accAddr := PrivKey()
+	return accAddr
+}
+
+// PrivKey returns a private key and corresponding on-chain address.
+func PrivKey() (*secp256k1.PrivKey, sdk.AccAddress) {
+	privKey := secp256k1.GenPrivKey()
+	pubKey := privKey.PubKey()
+	addr := pubKey.Address()
+	return privKey, sdk.AccAddress(addr)
+}
+
+// PrivKeyAddressPairs generates (deterministically) a total of n private keys and addresses.
+func PrivKeyAddressPairs(n int) (keys []cryptotypes.PrivKey, addrs []sdk.AccAddress) {
+	r := rand.New(rand.NewSource(12345)) // make the generation deterministic
+	keys = make([]cryptotypes.PrivKey, n)
+	addrs = make([]sdk.AccAddress, n)
+	for i := 0; i < n; i++ {
+		secret := make([]byte, 32)
+		_, err := r.Read(secret)
+		if err != nil {
+			panic("Could not read randomness")
+		}
+		keys[i] = secp256k1.GenPrivKeyFromSecret(secret)
+		addrs[i] = sdk.AccAddress(keys[i].PubKey().Address())
+	}
+	return
+}
+
+// Ed25519 functions as a wrapper class implementing the "AccAddress" and
+// "PrivKeyAddressPairs" methods using ed25519.
+var Ed25519 ed25519Algo = ed25519Algo{}
```

Comment:
why we add this?

Reply:
Just saw your description, but adding this without having really a full support for creating accounts using Ed25519 is out of the scope of the PR.

Reply:
I only added that to avoid deleting it entirely. I'm fine with removing it too

Reply:
5eea4a77e2276696b13a7d75fe1425693bfe0d23